core/adapters/http_test.go

```diff
@@ -121,7 +121,7 @@ func TestHTTPGet_TimeoutAllowsRetries(t *testing.T) {
 		require.NoError(t, err)
 		assert.Greater(t, len(b), 0)
 		attempts <- struct{}{}
-		timeoutOnce.Do(func() { time.Sleep(81 * time.Millisecond) })
+		timeoutOnce.Do(func() { time.Sleep(100 * time.Millisecond) })
```

Comment:
I'd be tempted to simply `t.Skip` this test since it's always going to be probabilistic to some extent no matter what we do here.

Reply:
Unless we somehow do dependency injection with a timer interface

Reply:
We could use build tags instead of `t.Skip` so that we could run these probabalistic tests separately

Reply:
@jkongie @samsondav ok - i moved it to a separate file with a build tag